### ISSUE TITLE
'this' reference from static method should be possible

### ISSUE THREAD

**Issue by aguzubiaga** [Microsoft/TypeScript#6331]
In JavaScript I am able to reference `this` from object methods, in order to access properties from the same object that stores my method, for example the prototype of the object.

Suppose I'm writing an ORM and I have a class `Model` and I want to implement an static method `where` that does a query in the DB for the corresponding table to that `Model`. I would write it like this:

```ts
class Model {
  public static where(options: any) {
     doWhereQuery(Reflect.getMetadata('table', this.prototype));
  }
}

 // Suppose that I have a decorator that sets the 'table' key in this class metadata
@table('users')
class User extends Model { }

User.where({ name: 'John', surname: 'Doe' });
```

But when I do so I get: `error TS2334: 'this' cannot be referenced in a static property initializer.`

I understand that maybe `this` is not the correct keyword to reference the "class", but that is the way it works in JavaScript and I think that this should be possible Otherwise, there is no trivial way to implement that functionality.

**Comment by RyanCavanaugh** [comment]
It is legal to reference `this` from a static method. The sample you posted does not produce the error given. Can you clarify?

**Comment by aguzubiaga** [comment]
Oh sorry, you are right. But it produces that error if you do it from an static arrow function and I guessed it would happen from an static method too. Anyway, that is incorrect, isn't it?

**Comment by RyanCavanaugh** [comment]
What's the intent with writing it as an arrow function *and* referencing `this` ?

If you want `this` to always refer to `Model`, then just write `Model` instead of `this`.

If you want `this` to refer to the class that it's invoked on *but not require a method invocation* (i.e. allow an unbound invocation `var x = MyModel.where; x();`, that's impossible because derived classes don't get a newly-closed-over own property for `where`, so there's no way for them to capture `this`. Trying to write out what the emit would look like in this case is an informative exercise.

**Comment by aguzubiaga** [comment]
I forgot that arrow functions doesn't have own this, I would close the issue. Sorry for my ignorance.
